File: 22-Strategy/src/go/paypal_strategy.go
package main

import "fmt"

// PayPalStrategy is a Concrete Strategy — pays via PayPal.
type PayPalStrategy struct {
	email string
}

func (p *PayPalStrategy) Pay(amount float64) {
	fmt.Printf("  Paid $%.2f via PayPal account %s\n", amount, p.email)
}

File: 22-Strategy/src/rust/paypal_strategy.rs
use crate::payment_strategy::PaymentStrategy;

/// Concrete Strategy — pays via PayPal.
pub struct PayPalStrategy {
    email: String,
}

impl PayPalStrategy {
    pub fn new(email: &str) -> Self {
        PayPalStrategy { email: email.to_string() }
    }
}

impl PaymentStrategy for PayPalStrategy {
    fn pay(&self, amount: f64) {
        println!("  Paid ${:.2} via PayPal account {}", amount, self.email);
    }
}
